### PR TITLE
Fix addGood bug

### PR DESCRIPTION
**Pull request by stephaneleong**
filter out goods without an exisiting suppliername #124 
Csv will be filtered based on initial load
filtered will be applied when goods are added using the addgood command

done at a Model level because it needs access to the addressbook

### CODE REVIEW

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 1]
```diff
@@ -165,12 +170,32 @@ public void getGoods_modifyList_throwsUnsupportedOperationException() {
...
+    @Test
+    public void addGoods_illegalSupplierName_throwsIllegalSupplierName() {
+        Person person = new PersonBuilder().build();
+        modelManager.addPerson(person);
+        GoodsReceipt goodsReceipt = new GoodsReceiptBuilder()
+                .withSupplierName(new Name("Invalid Name"))
+                .build();
+
+        assertFalse(person.getName().equals(goodsReceipt.getSupplierName()));
+        Assertions.assertThrows(IllegalSupplierNameException.class, () -> modelManager.addGoods(goodsReceipt));
```
Comment (jialegitgud) [inline src/test/java/seedu/address/model/ModelManagerTest.java:185]: By convention, import only `assertThrows` and not the whole Assertions library.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -223,4 +231,15 @@ public int getFilteredGoodsQuantityStatistics() {
     public double getFilteredGoodsCostStatistics() {
         return GoodsReceiptUtil.sumTotals(filteredReceipts);
     }
+
+    private boolean hasSupplierName(GoodsReceipt goodsReceipt) {
```
Comment (jialegitgud) [inline src/main/java/seedu/address/model/ModelManager.java:235]: Naming could be changed to better reflect what it does, a suggestion being `checkExistingSupplier`

**Review by jialegitgud — COMMENTED** [review]
Overall, LGTM 👍

File: src/test/java/seedu/address/model/ModelManagerTest.java (hunk 1 above)
Comment (stephaneleong) [reply src/test/java/seedu/address/model/ModelManagerTest.java]: fix coding styles

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (stephaneleong) [reply src/main/java/seedu/address/model/ModelManager.java]: changed it to hasExistingSupplier instead

**Pull request merged**